parsing/dialects/dialect1/parserImpls.ftl

```diff
@@ -1999,3 +1999,35 @@ int VariableBinaryTypePrecision() :
     <RPAREN>
     { return precision; }
 }
+
+SqlTypeNameSpec SqlPeriodDataType() :
+{
+    final TimeScale timeScale;
+    SqlNumericLiteral precision = null;
+    boolean isWithTimezone = false;
+}
+{
+    <PERIOD> <LPAREN>
+    (
+        <DATE> { timeScale = TimeScale.DATE; }
+    |
+        (
+            <TIME> { timeScale = TimeScale.TIME; }
+        |
+            <TIMESTAMP> { timeScale = TimeScale.TIMESTAMP; }
+        )
+        [
+            <LPAREN>
+            precision = UnsignedNumericLiteral()
+            <RPAREN>
+        ]
+        [
+            <WITH> <TIME> <ZONE> { isWithTimezone = true; }
+        ]
+    )
+    <RPAREN>
+    {
+        return new SqlPeriodTypeNameSpec(timeScale, precision, isWithTimezone,
+        getPos());
```

Comment:
nit: this line's indentation looks off

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/177/commits/110d83e563426a9ea05113263c6efc02c44447dc